src/pages/task/taskTabs/logs/LogTypes.tsx

```diff
@@ -174,32 +172,20 @@ const useRenderBody: React.VFC<{
   loading: boolean;
   error: ApolloError;
   data: (TaskEventLogEntryType | LogMessageType)[];
-  currentLog: LogTypes;
   LogContainer?: React.VFC<{ children: React.ReactNode }>;
-  htmlLink: string;
-  rawLink: string;
-  lobsterLink: string;
+  setNoLogs: (noLogs: boolean) => void;
 }> = ({
   loading,
   error,
   data,
-  currentLog,
-  rawLink,
-  htmlLink,
-  lobsterLink,
   LogContainer = ({ children }) => <StyledPre>{children}</StyledPre>,
+  setNoLogs,
 }) => {
-  const taskAnalytics = useTaskAnalytics();
-  const updateQueryParams = useUpdateURLQueryParams();
   const noLogs = !!((error && !data) || !data.length);
-  const onChangeLog = (event: React.ChangeEvent<HTMLInputElement>): void => {
-    const nextLogType = event.target.value as LogTypes;
-    updateQueryParams({ [QueryParams.LogType]: nextLogType });
-    taskAnalytics.sendEvent({
-      name: "Select Logs Type",
-      logsType: nextLogType,
-    });
-  };
+  // Update the value of noLogs in the parent component.
+  useEffect(() => {
+    setNoLogs(noLogs);
+  }, [setNoLogs, noLogs]);
```

Comment:
All of the information needed to calculate `noLogs` is present in the parent component, might it make more sense to set that variable there rather than in this hook?

Reply:
I think I need the results of the query to determine if there are no new logs (i.e. `data_from_query.length === 0`) 😔 It could also be done through Apollo's `onCompleted` or `onError`, though that becomes a little verbose 😅

I also feel like maybe it could be done through a larger refactor 🤔 I explored it a little but it was too big for this PR. I will think about it more and maybe open a ticket for it

Reply:
Oh yes I must've misread! sorry, sgtm 👍